src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -436,13 +436,10 @@ private int getAndSetNetworkTimeout(final Connection connection, final long time
             if (isNetworkTimeoutSupported == UNINITIALIZED) {
                isNetworkTimeoutSupported = FALSE;
 
-               LOGGER.warn("{} - Unable to get/set network timeout for connection. ({})", poolName, e.getMessage());
+               LOGGER.warn("{} - Failed to get/set network timeout for connection. ({})", poolName, e.getMessage());
                if (validationTimeout < SECONDS.toMillis(1)) {
                   LOGGER.warn("{} - A validationTimeout of less than 1 second cannot be honored on drivers without setNetworkTimeout() support.", poolName);
                }
-               else if (validationTimeout % SECONDS.toMillis(1) != 0) {
-                  LOGGER.warn("{} - A validationTimeout with fractional second granularity cannot be honored on drivers without setNetworkTimeout() support.", poolName);
-               }
```

Comment:
A few reasons.

1) Users do not know what APIs that HikariCP uses internally for validation.  We use ``isValid()`` or ``setQueryTimeout()``, but we should not expect that the user should know.

2) Because we also use ``setNetworkTimeout()`` when it is available, we can support a ``validationTimeout`` with fractional seconds, for example 1250ms.  If the user sets a ``validationTimeout`` of 1250ms, *and* the driver supports ``setNetworkTimeout()`` we can actually honor the 1250ms timeout even though we call ``isValid()`` with 2 seconds as a timeout.  If the driver *does not* support ``setNetworkTimeout()`` the we cannot honor the user's requested 1250ms timeout, and therefore the warning is useful.

If we log this warning, the user is less likely to open an issue that HikariCP is not honoring their ``validationTimeout``.